x-pack/plugins/security_solution/public/management/pages/endpoint_hosts/view/details/endpoint_activity_log.tsx

```diff
@@ -19,54 +27,95 @@ import {
   getActivityLogError,
   getActivityLogIterableData,
   getActivityLogRequestLoaded,
+  getLastLoadedActivityLogData,
   getActivityLogRequestLoading,
 } from '../../store/selectors';
 
+const Sentinel = styled.div`
+  height: 6px;
+  width: 100%;
+`;
+
 export const EndpointActivityLog = memo(
   ({ activityLog }: { activityLog: AsyncResourceState<Immutable<ActivityLog>> }) => {
     const activityLogLoading = useEndpointSelector(getActivityLogRequestLoading);
     const activityLogLoaded = useEndpointSelector(getActivityLogRequestLoaded);
+    const activityLastLogData = useEndpointSelector(getLastLoadedActivityLogData);
     const activityLogData = useEndpointSelector(getActivityLogIterableData);
+    const activityLogSize = activityLogData.length;
     const activityLogError = useEndpointSelector(getActivityLogError);
-    const dispatch = useDispatch<(a: EndpointAction) => void>();
-    const { page, pageSize } = useEndpointSelector(getActivityLogDataPaging);
+    const dispatch = useDispatch<(action: EndpointAction) => void>();
+    const { page, pageSize, disabled: isPagingDisabled } = useEndpointSelector(
+      getActivityLogDataPaging
+    );
+
+    const fetchMoreSentinel = useRef<HTMLInputElement | null>(null);
+    const getActivityLog = useCallback(
+      (entries: IntersectionObserverEntry[]) => {
+        const isIntersecting = entries.some((entry) => entry.intersectionRatio > 0);
+        if (isIntersecting && activityLogLoaded && !isPagingDisabled) {
+          dispatch({
+            type: 'appRequestedEndpointActivityLog',
+            payload: {
+              page: page + 1,
+              pageSize,
+            },
+          });
+        }
+      },
+      [activityLogLoaded, dispatch, isPagingDisabled, page, pageSize]
+    );
 
-    const getActivityLog = useCallback(() => {
-      dispatch({
-        type: 'appRequestedEndpointActivityLog',
-        payload: {
-          page: page + 1,
-          pageSize,
-        },
+    useEffect(() => {
+      const observer = new IntersectionObserver(getActivityLog, {
```

Comment:
Does the `observer` need to be a new instance every time? or can it be initialized only once (ex. via a `useMemo()` above) and then just `.observe()`/`.disconnect()`?

Reply:
@paul-tavares Yeah, I see what you mean, although here the observer is instance is created only when the dependency changes which is the callback method (`getActivitLog`), and React guarantees that the callback method doesn't change. 

Moving the instantiation within a `useMemo` above will still need the callback function as a dependency as we need it while creating the observer instance. Not sure if it's worth doing the change. 🤔

Reply:
👍